tests/MimeTest.php

```diff
@@ -51,15 +51,19 @@ public function testWebp()
 
     public function testDetermineDimensions()
     {
-        $this->assertSame([16, 16], $this->detector->determineDimensions(__DIR__ . '/files/sample.png'));
+        $file = new \SplFileObject(__DIR__ . '/files/sample.png', 'r');
+
+        $this->assertSame([16, 16], $this->detector->determineDimensions($file));
     }
 
     public function testDetermineDimensionsWithInvalidImagePath()
     {
-        $this->expectException(\InvalidArgumentException::class);
-        $this->expectExceptionMessage('File path invalid_image_file is not existed');
+        $this->expectException(\RuntimeException::class);
+        $this->expectExceptionMessage('failed to open stream: No such file or directory');
+
+        $file = new \SplFileObject('invalid_path.png', 'r');
```

Comment:
This test does nothing. The Exception you are testing for is thrown by [`SplFileObject::__construct`](https://secure.php.net/manual/en/splfileobject.construct.php), not anything done by the detector.

The beauty of requiring a valid [`SplFileObject`](https://secure.php.net/manual/en/class.splfileobject.php) is that we do not need to care about this, by the time the object gets passed to `determineDimensions` we can already be sure that it is valid.

Reply:
You're right. I will remove that test :+1:.